### DISCUSSION TITLE
GlobalParameter expression evaluation/execution

### DISCUSSION THREAD

**Discussion by bmasotta-lcg** [microsoft/RulesEngine#476] [Q&A]
I found that GlobalParameter expressions are evaluated (executed) one time for each rule, no matter if that rule uses it or not
This can lead to problems when GlobalParameters are tied to calculations (performance problems) or to operations that can generate side effects

I need to use global parameters for parameters that are used in more than one rule

Is There a way for changing this behavior?

Thanks a lot

**Comment by Bugattisport1** [comment]
Also facing the same issue... is there any way to have the global parameters evaluate once and then have each rule just use the result?

**Comment by shokri-navid** [comment]
@abbasc52 
can you provide more information about this topic? 
I have the same issue [there](https://github.com/microsoft/RulesEngine/discussions/86#discussioncomment-10242314)
